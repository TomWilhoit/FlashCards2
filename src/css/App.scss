$font-title: "Roboto Condensed", sans-serif;
$font-copy: "Lato", sans-serif;
$header-font: "Abel", sans-serif;
$font-color: #fff;
$btn-color: #160f42;
$bckg-color: #0c3e9c;

* {
  box-sizing: border-box;
}

html {
  height: 100%;
}

body {
  height: 100%;
  margin: 0;
  font-family: $font-copy;
}

#root {
  height: 100%;
}

.App {
  text-align: center;
  background-image: url(./images/splash2.jpeg);
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: bottom center;

  .header {
    padding: 2em 0;
    background: #000000b3;
    color: #fff;
    box-shadow: -1px 3px 15px 4px rgba(0, 0, 0, 0.4);

    h1 {
      margin: 0;
      font-family: $header-font;
      font-size: 2.5em;
    }
  }
}

.congrats-page {
  background-image: url(./images/splash2.jpeg);
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: bottom center;
}

.percentage {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 45px;
  color: lime;
  font-weight: 100;
  font-family: $font-copy;
}

.question-counter {
  display: flex;
  justify-content: space-around;
  color: $font-color;
  font-size: 25px;
  padding: 2em 0;
  font-family: $header-font;
}

.restart-game-btn {
  margin: auto;
  display: -webkit-flex;
  display: flex;
  width: 19%;
  /* height: 20%; */
  -webkit-justify-content: center;
  justify-content: center;
  padding: 10px;
  background: lime;
  color: black;
  font-size: 25px;
  border-radius: 5px;
  border-style: none;
  min-width: 100px;
}

.review-header {
  text-align: center;
}

.win-message {
  font-size: 60px;
  text-align: center;
  font-family: $header-font;
  color: $font-color;
  margin: 25px;
  background: #292727;
}

.win-percentage {
  font-size: 60px;
  text-align: center;
  color: lime;
  padding: 42px;
}

.wrong-list-cont {
  background: #292727;
  max-height: 70%;
  overflow: scroll;
}

.wrong-questions {
  display: flex;
  justify-content: center;
  flex-direction: column;
  background-image: url(./images/splash2.jpeg);
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: bottom center;
  color: $font-color;
  font-weight: 100;
  font-family: $font-copy;
}
