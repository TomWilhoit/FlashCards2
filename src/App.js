import React, { Component } from "react";
import Header from "./Header";
import CardContainer from "./CardContainer";
import Card from "./Card";
import mockData from "./mockData.js";
import "./css/App.css";

class App extends Component {
  constructor() {
    super();
    this.state = {
      questions: mockData.mockData,
      gameRestart: false,
      repeatQuestions: true,
      questionIndex: 0,
      savedArray: [],
      wrongArray: []
    };
  }

  componentWillMount() {
    const localStorageArray = JSON.parse(
      localStorage.getItem("savedQuestions")
    );
    console.log(localStorageArray);
    this.setState({
      wrongArray: localStorageArray
    });
    console.log(this.state.wrongArray);
  }

  shouldRepeatQuestions = () => {
    if (this.state.repeatQuestions === false) {
      this.setState({
        repeatQuestions: true
      });
    } else {
      this.setState({
        repeatQuestions: false
      });
    }
  };

  shouldRepeatQuestions = () => {
    if (this.state.questionIndex === 55) {
      this.setState({
        repeatQuestions: true
      });
    }
  };

  incrementQuestionIndex = () => {
    const newIndex = this.state.questionIndex + 1;
    this.setState({
      questionIndex: newIndex
    });
  };

  saveToStorage = () => {
    let [...savedQuestions] = this.state.savedArray;
    savedQuestions.push(this.state.questions[this.state.questionIndex]);
    localStorage.setItem("savedQuestions", JSON.stringify(savedQuestions));
    this.setState({
      savedArray: savedQuestions
    });
  };

  renderWrongArray = () => {
    console.log(this.state.wrongArray);
    return this.state.wrongArray.map((element, index) => {
      return (
        <ul>
          <li key={index}>{element.id}</li>
          <li key={index}>{element.question}</li>
          <li key={index}>{element.correctAnswer}</li>
        </ul>
      );
    });
  };

  render() {
    if (this.state.repeatQuestions === false) {
      return (
        <div className="App">
          <Header />
          <CardContainer />
          <Card
            questions={this.state.questions[this.state.questionIndex]}
            questionIndex={this.state.questionIndex}
            incrementQuestionIndex={this.incrementQuestionIndex}
            saveToStorage={this.saveToStorage}
            shouldRepeatQuestions={this.shouldRepeatQuestions}
          />
        </div>
      );
    } else {
      return (
        <div className="App">
          <div>{this.renderWrongArray()}</div>
        </div>
      );
    }
  }
}

export default App;
