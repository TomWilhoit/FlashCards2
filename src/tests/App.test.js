import React from "react";
import App from "../App";
import { shallow } from "enzyme";
global.localStorage = require("../setupTests");

describe("App", () => {
  let wrapper;
  const mockFunc = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App restartGame={mockFunc} />);
  });

  // State and Snapshot tests

  it("should have a proper default state", () => {
    expect(wrapper.state()).toEqual({
      questions: [],
      gameRestart: false,
      repeatQuestions: false,
      questionIndex: 0,
      savedArray: [],
      wrongArray: []
    });
  });

  it("should match the snapshot with all data passed in correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // restartGame method test

  it("should change the state of repeatQuestions when restartGame is invoked", () => {
    expect(wrapper.state("repeatQuestions")).toEqual(false);
    wrapper.instance().restartGame();
    expect(wrapper.state("repeatQuestions")).toEqual(true);
  });

  // shouldRepeatQuestions method test

  it("should invoke restartGame when shouldRepeatQuestions is invoked with proper state", () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, "restartGame");
    wrapper.setState({ questionIndex: 56 });
    wrapper.instance().shouldRepeatQuestions();
    expect(instance.restartGame).toBeCalled();
  });

  // closeOutGame method tests

  it("should invoke clearLocalStorage when closeOutGame is invoked", () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, "clearLocalStorage");
    wrapper.instance().closeOutGame();
    expect(instance.clearLocalStorage).toBeCalled();
  });

  it("should change state properties when closeOutGame is invoked", () => {
    wrapper.setState({
      gameRestart: true,
      repeatQuestions: true,
      questionIndex: 1,
      savedArray: [1, 2],
      wrongArray: [1, 2]
    });
    wrapper.instance().closeOutGame();
    expect(wrapper.state("gameRestart")).toEqual(false);
    expect(wrapper.state("repeatQuestions")).toEqual(false);
    expect(wrapper.state("questionIndex")).toEqual(0);
    expect(wrapper.state("savedArray")).toEqual([]);
    expect(wrapper.state("wrongArray")).toEqual([]);
  });

  // incrementQuestionIndex

  it("should change the state of questionIndex when incrementQuestionIndex is invoked", () => {
    expect(wrapper.state("questionIndex")).toEqual(0);
    wrapper.instance().incrementQuestionIndex();
    expect(wrapper.state("questionIndex")).toEqual(1);
  });
});
