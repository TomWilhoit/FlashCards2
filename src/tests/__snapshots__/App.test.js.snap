// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div
  className="loading-screen"
>
  Loading
</div>
`;
